src/main/java/gyro/google/compute/AbstractDiskResource.java

```diff
@@ -0,0 +1,293 @@
+        if (labels == null) {
+            labels = new HashMap<>();
+        }
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * The physical block size of the disk, in bytes. Valid values are ``4096`` or ``16384``. Defaults to ``4096``.
+     */
+    @ValidNumbers({ 4096, 16384 })
+    public Long getPhysicalBlockSizeBytes() {
+        if (physicalBlockSizeBytes == null) {
+            physicalBlockSizeBytes = 4096L;
+        }
+
+        return physicalBlockSizeBytes;
+    }
+
+    public void setPhysicalBlockSizeBytes(Long physicalBlockSizeBytes) {
+        this.physicalBlockSizeBytes = physicalBlockSizeBytes;
+    }
+
+    /**
+     * The status of disk creation. Values can be: ``CREATING``, ``RESTORING``, ``FAILED``, ``READY``, or ``DELETING``.
+     */
+    @Output
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+
+    /**
+     * The unique ID of the snapshot used to create the disk.
+     */
+    @Output
+    public String getSourceSnapshotId() {
+        return sourceSnapshotId;
+    }
+
+    public void setSourceSnapshotId(String sourceSnapshotId) {
+        this.sourceSnapshotId = sourceSnapshotId;
+    }
+
+    /**
+     * The fully-qualified URL linking back to the disk.
+     */
+    @Output
+    @Id
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink != null ? toDiskUrl(getProjectId(), selfLink) : null;
+    }
+
+    /**
+     * Links to the attached instances of the disk.
+     */
+    @Output
+    public List<String> getUsers() {
+        if (users == null) {
+            users = new ArrayList<>();
+        }
+        return users;
+    }
+
+    public void setUsers(List<String> users) {
+        this.users = users;
+    }
+
+    /**
+     * The fingerprint for the labels being applied to the disk, which is updated any time the labels change.
+     */
+    @Output
+    public String getLabelFingerprint() {
+        return labelFingerprint;
+    }
+
+    public void setLabelFingerprint(String labelFingerprint) {
+        this.labelFingerprint = labelFingerprint;
+    }
+
+    @Override
+    public void copyFrom(Disk disk) {
+        setName(disk.getName());
+        setDescription(disk.getDescription());
+        setSizeGb(disk.getSizeGb());
+        setSourceSnapshot(findById(SnapshotResource.class, disk.getSourceSnapshot()));
+        setLabels(disk.getLabels());
+        setPhysicalBlockSizeBytes(disk.getPhysicalBlockSizeBytes());
+        setStatus(disk.getStatus());
+        setSourceSnapshotId(disk.getSourceSnapshotId());
+        setSelfLink(disk.getSelfLink());
+        setUsers(disk.getUsers());
+        setLabelFingerprint(disk.getLabelFingerprint());
+
+        if (disk.getDiskEncryptionKey() != null) {
```

Comment:
This ensure that the value would be null if not present than using the value from the state.
```suggestion
        setDiskEncryptionKey(null);
        if (disk.getDiskEncryptionKey() != null) {
```